### PR TITLE
AVL tree in Javascript

### PR DESCRIPTION
**Pull request by lipsamishra1609**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

Add AVL Tree.
AVL tree is a self-balancing Binary Search Tree (BST) where the difference between heights of left and right subtrees cannot be more than one for all nodes.   It was the first such data structure to be invented.
 
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by atarax665** [comment]
Fill the PR template.

File: JavaScript/ds/AVLTree.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+class AVLTree {
+   constructor() {
+
```
Comment (vatsalkesarwani12) [inline JavaScript/ds/AVLTree.js]: remove these extra spaces

Comment (lipsamishra1609) [reply JavaScript/ds/AVLTree.js]: Done

**Comment by vatsalkesarwani12** [comment]
resolve conflict

**Comment by lipsamishra1609** [comment]
resolved

**Review by vatsalkesarwani12 — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]
LGTM

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @lipsamishra1609 for code

**Pull request merged**